cmake/ElektraCompiling.cmake

```diff
@@ -61,10 +61,27 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 
 	if (ENABLE_DEBUG)
 		set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=undefined -fsanitize=integer")
-		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=undefined -fsanitize=integer")
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lubsan")
 	endif()
 endif()
 
+if (ENABLE_ASAN)
+	set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+	set (ASAN_LIBRARY "-lasan") #this is needed for GIR to put asan in front
+	if (CMAKE_COMPILER_IS_GNUCXX)
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
+		# this is currently a workaround so gtests compile with ASAN
+		#TODO someone with expertise migrate this into LibAddTest
+		set (COMMON_FLAGS "${COMMON_FLAGS} -lpthread")
```

Comment:
I was asking for expertise in regard of -lpthread in combination of the gtests as this was the place i got [problems](https://github.com/ElektraInitiative/libelektra/pull/803#issuecomment-226262269) with the linking on gcc. As some internet posts suggest this could be a problem of linking flag order. Using it for general linking surely solves the problem but a more granular solution would have been nicer.